### PR TITLE
Effect 2

### PR DESCRIPTION
**Pull request by ArchDemons**

### CODE REVIEW

File: src/toniarts/openkeeper/view/KeeperHand.java [hunk 1]
```diff
@@ -188,7 +189,9 @@ private void removeIcon() {
     }
 
     private void addIcon(final ArtResource image, final int index) {
-        if (!(image.getSettings() instanceof ArtResource.Image)) {
+        if (image.getType() != ArtResourceType.ALPHA
+                || image.getType() != ArtResourceType.ADDITIVE_ALPHA
```
Comment (ufdada) [inline src/toniarts/openkeeper/view/KeeperHand.java]: I think there should be an && operator

File: src/toniarts/openkeeper/view/KeeperHand.java [hunk 2]
```diff
@@ -188,7 +189,9 @@ private void removeIcon() {
...
+                || image.getType() != ArtResourceType.SPRITE) {
```
Comment (ufdada) [inline src/toniarts/openkeeper/view/KeeperHand.java]: and here

**Comment by ArchDemons** [comment]
Known bug with torch flame. It's not in correct position (I do not know how to fix it). Need change Material of Flame from current to Particle (problem with background black color), but when I change, I got black rectangle...

**Comment by tonihele** [comment]
It is nice though, our torches aren't yet in right places anyway :) Only troubles me that the overall atmosphere in our dungeon doesn't reflect the original :) I'll try to look it closer tomorrow, I tried it already briefly yesterday.

We have a fundamental problem with the placing. We need to reset all models somehow, feels wrong, maybe a simple thing but I never studied it further. Maybe causing all sorts of problems down the line.

File: src/toniarts/openkeeper/tools/convert/map/Trigger.java [hunk 3]
```diff
@@ -32,38 +32,38 @@
...
+            data = new HashMap<>();
         }
 
-        if (data == null) {
-            userData.remove(key);
-        } else if (data instanceof Number) {
-            userData.put(key, data);
+        if (value == null) {
+            data.remove(key);
+        } else if (value instanceof Number) {
```
Comment (tonihele) [inline src/toniarts/openkeeper/tools/convert/map/Trigger.java]: Same here?

File: src/toniarts/openkeeper/tools/convert/map/ArtResource.java [hunk 4]
```diff
@@ -93,219 +100,63 @@ protected void setName(String name) {
...
-            return id;
+    public Collection<String> getDataKeys() {
+        if (data != null) {
+            return data.keySet();
         }
 
-        protected void setId(int id) {
-            this.id = id;
-        }
+        return Collections.EMPTY_SET;
```
Comment (tonihele) [inline src/toniarts/openkeeper/tools/convert/map/ArtResource.java]: I believe one should use the methods instead, emptySet()

File: src/toniarts/openkeeper/tools/convert/map/Trigger.java (hunk 3 above)
Comment (ArchDemons) [inline src/toniarts/openkeeper/tools/convert/map/Trigger.java]: it`s always number

**Comment by tonihele** [comment]
Nice nice

**Pull request merged**